astropy/modeling/fitting.py

```diff
@@ -742,9 +743,14 @@ def __call__(self, model, x, y, z=None, weights=None,
         from scipy import optimize
 
         model_copy = _validate_model(model, self.supported_constraints)
-        farg = (model_copy, weights, ) + _convert_input(x, y, z)
-
-        if model_copy.fit_deriv is None or estimate_jacobian:
+        n_models = len(model_copy)
+        model_set_axis = False if n_models > 1 else model_copy.model_set_axis
+        farg = ((model_copy, weights, model_set_axis) +
+                _convert_input(x, y, z, n_models=n_models, rollaxis=False,
+                               model_set_axis=model_set_axis))
```

Comment:
I think `model_set_axis` here does not mean the same thing as in `objective_function` (you're passing the same variable to both)... There you are specifying the axis you want to use when evaluating the model with `__call__` (ie. `False`) whereas here you are communicating which axis the model was actually instantiated with (ie. `model_copy.model_set_axis`). That's my understanding anyway. This implicitly assumes that your `y`/`z` input array matches the model set definition, but AFAICT that's how it's intended to work.